featurebyte/config.py

```diff
@@ -271,37 +271,53 @@ class Configurations:
         skipped_members=["get_client"],
     )
 
-    def __init__(self, config_file_path: Optional[str] = None) -> None:
+    _instance = None
+
+    def __new__(cls, *args: Any, **kwargs: Any) -> "Configurations":
+        if Configurations._instance is None:
+            cls._instance = super().__new__(cls)
+        return cls._instance
+
+    def __init__(self, config_file_path: Optional[str] = None, force: bool = False) -> None:
         """
         Load and parse configurations
 
         Parameters
         ----------
         config_file_path: str | None
             Path to read configurations from
-        """
-        home_path = get_home_path()
-        self._config_file_path = (
-            Path(config_file_path) if config_file_path else home_path.joinpath("config.yaml")
-        )
-
-        # create config file if it does not exist
-        if not self._config_file_path.exists():
-            self._config_file_path.parent.mkdir(parents=True, exist_ok=True)
-            self._config_file_path.write_text(
-                "# featurebyte configuration\n\n"
-                "profile:\n"
-                "  - name: local\n"
-                "    api_url: http://127.0.0.1:8088\n\n"
-                "default_profile: local\n\n"
+            this will force a reload of configuration if specified
+        force: bool
+            Force reload configurations
```

Comment:
Would it be a bit clearer to name this parameter as `reload=True`?

Reply:
Configurations("/config.yaml" , reload=False) will still reload. I think that might be mote confusing